R/messages.R

```diff
@@ -81,5 +81,18 @@ setvar <- function(core, dates, var, values, unit)
     unit[is.na(unit)] <- '(unitless)'
     sendmessage(core, SETDATA(), var, dates, values, unit)
 
+    if(any(dates <= getdate(core)) || any(is.na(dates))) {
+        rdate <- min(dates) -1
+        if(is.na(rdate))
+            rdate <- 0
+
+        if(core$clean)
+            core$reset_date <- rdate
+        else
+            core$reset_date <- min(rdate, core$reset_date)
+
+        core$clean <- FALSE
+    }
+
```

Comment:
Not I defer to you (all) as the more experienced coders, but FWIW, both the [Google](https://google.github.io/styleguide/Rguide.xml#curlybraces) and [Tidyverse](https://style.tidyverse.org/syntax.html#indenting) R style guides recommend using braces around `if-else` statements, even if they are not technically required for single lines.

Both those style guides also suggest spaces after `if` (as in `if (condition)` rather than `if(condition)`).

Personally, I'm also generally wary of using `$` because it does partial matching. So, if we changed `clean` to `clean_core` at some later point, `core$clean` would still work. For internal code like this, it probably shouldn't matter, but I've been bitten by this multiple times in the PEcAn code base and have been wary of it ever since.

Reply:
I noticed the brace thing too but thought I was being overly picky. Note there are cases where this makes a difference, e.g. I don't think `covr` can compute code coverage for the two cases without them.